Scaffolding Broken for quarkus-messaging-kafka

### Describe the bug

When I scaffold the following project (with Java 25):

```
mvn io.quarkus:quarkus-maven-plugin:create \
  -DprojectGroupId=com.example \
  -DprojectArtifactId=review-workflow \
  -Dextensions="rest-jackson,messaging-kafka,quarkus-flow,quarkus-langchain4j-ollama"
```

The resulting artifacts show errors: 
in src/main/java/com/example
- MyMessagingApplication.java (The type Message is not generic; it cannot be parameterized with arguments <String>)

in src/main/test/java/com/example
- MyMessagingApplicationTest (The method toUpperCase(Message.of("bonjour")) is undefined for the type MyMessagingApplication)


### Expected behavior

Code compiles.

### Actual behavior

[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /Users/meisele/Projects/the-main-thread/review-workflow/src/main/java/com/example/MyMessagingApplication.java:[32,47] type com.example.Message does not take parameters
[ERROR] /Users/meisele/Projects/the-main-thread/review-workflow/src/main/java/com/example/MyMessagingApplication.java:[32,19] type com.example.Message does not take parameters
[INFO] 2 errors 
[INFO] ----------

### How to Reproduce?

See above. Scaffold project.

### Output of `uname -a` or `ver`

Darwin Kernel Version 25.2.0: Tue Nov 18 21:09:56 PST 2025; root:xnu-12377.61.12~1/RELEASE_ARM64_T6041 arm64

### Output of `java -version`

openjdk 25 2025-09-16 LTS OpenJDK Runtime Environment Temurin-25+36 (build 25+36-LTS)

### Quarkus version or git rev

3.31.3

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

Apache Maven 3.9.11 (3e54c93a704957b63ee3494413a2b544fd3d825b)

### Additional information

_No response_

--- a/devtools/project-core-extension-codestarts/src/main/resources/codestarts/quarkus/extension-codestarts/messaging-codestart/java/src/main/java/org/acme/MyMessagingApplication.java
+++ b/devtools/project-core-extension-codestarts/src/main/resources/codestarts/quarkus/extension-codestarts/messaging-codestart/java/src/main/java/org/acme/MyMessagingApplication.java
@@ -1,17 +1,21 @@
 package org.acme;
 
 import io.quarkus.runtime.StartupEvent;
-import org.eclipse.microprofile.reactive.messaging.*;
-
 import jakarta.enterprise.context.ApplicationScoped;
 import jakarta.enterprise.event.Observes;
-import jakarta.inject.Inject;
+import org.eclipse.microprofile.reactive.messaging.Channel;
+import org.eclipse.microprofile.reactive.messaging.Emitter;
+import org.eclipse.microprofile.reactive.messaging.Incoming;
+import org.eclipse.microprofile.reactive.messaging.Outgoing;
+
 import java.util.stream.Stream;
 
 @ApplicationScoped
 public class MyMessagingApplication {
 
-    @Inject
+    /**
+     * Injects an emitter to send messages to the "words-out" channel.
+     */
     @Channel("words-out")
     Emitter<String> emitter;
 
@@ -25,16 +29,16 @@ public class MyMessagingApplication {
 
     /**
      * Consume the message from the "words-in" channel, uppercase it and send it to the uppercase channel.
-     * Messages come from the broker.
+     * This method is called by the framework when a message is received on the "words-in" channel (from the broker).
      **/
     @Incoming("words-in")
     @Outgoing("uppercase")
-    public Message<String> toUpperCase(Message<String> message) {
-        return message.withPayload(message.getPayload().toUpperCase());
+    public String toUpperCase(String message) {
+        return message.toUpperCase();
     }
 
     /**
-     * Consume the uppercase channel (in-memory) and print the messages.
+     * Consume the uppercase channel (coming from within the application) and print the messages to the console.
      **/
     @Incoming("uppercase")
     public void sink(String word) {

--- a/devtools/project-core-extension-codestarts/src/main/resources/codestarts/quarkus/extension-codestarts/messaging-codestart/java/src/test/java/org/acme/MyMessagingApplicationTest.java
+++ b/devtools/project-core-extension-codestarts/src/main/resources/codestarts/quarkus/extension-codestarts/messaging-codestart/java/src/test/java/org/acme/MyMessagingApplicationTest.java
@@ -1,10 +1,7 @@
 package org.acme;
 
-import io.quarkus.test.common.QuarkusTestResource;
-import io.quarkus.test.common.QuarkusTestResourceLifecycleManager;
 import io.quarkus.test.junit.QuarkusTest;
 
-import org.eclipse.microprofile.reactive.messaging.Message;
 import org.junit.jupiter.api.Test;
 
 import jakarta.inject.Inject;
@@ -19,7 +16,7 @@ class MyMessagingApplicationTest {
 
     @Test
     void test() {
-        assertEquals("HELLO", application.toUpperCase(Message.of("Hello")).getPayload());
-        assertEquals("BONJOUR", application.toUpperCase(Message.of("bonjour")).getPayload());
+        assertEquals("HELLO", application.toUpperCase("Hello"));
+        assertEquals("BONJOUR", application.toUpperCase("bonjour"));
     }
 }

